### PR TITLE
feat(FX-4599): add collectionsConnection to artwork field

### PR DESCRIPTION
**Pull request by nickskalkin**
# Request

```graphql
{
	artwork(id: "rodney-smith-pears-clinton-connecticut") {
		id

		collectionsConnection(first: 2) {
			totalCount

			edges {
				node {
					internalID
					name
					default
					saves
					
					artworksConnection(first: 1) {
						totalCount
					}
				}
			}
		}
	}
}
```

# Response

```graphql
{
	"data": {
		"artwork": {
			"id": "QXJ0d29yazo2M2RhZGI1Y2Y3MDg0MTAwMGI3YWY2Njc=",
			"collectionsConnection": {
				"totalCount": 2,
				"edges": [
					{
						"node": {
							"internalID": "ce7852f8-da4a-4c7f-be98-9d77970549b5",
							"name": "Photography",
							"default": false,
							"saves": true,
							"artworksConnection": {
								"totalCount": 1
							}
						}
					},
					{
						"node": {
							"internalID": "2fc5f80e-759f-4fbf-90a1-5a7f58785921",
							"name": "Saved Artwork",
							"default": true,
							"saves": false,
							"artworksConnection": {
								"totalCount": 81
							}
						}
					}
				]
			}
		}
	},
	"extensions": {
		"requests": {
			"gravity": {
				"requests": {
					"artwork/rodney-smith-pears-clinton-connecticut": {
						"time": "0s 291.832ms",
						"cache": false,
						"length": "0 B"
					},
					"collections?artwork_id=rodney-smith-pears-clinton-connecticut&offset=0&private=true&size=2&total_count=true&user_id=...": {
						"time": "0s 240.088ms",
						"cache": false,
						"length": "0 B"
					},
					"collection/2fc5f80e-759f-4fbf-90a1-5a7f58785921/artworks?page=1&private=true&size=1&sort=-created_at&total_count=true&user_id=...": {
						"time": "0s 415.462ms",
						"cache": false,
						"length": "0 B"
					},
					"collection/ce7852f8-da4a-4c7f-be98-9d77970549b5/artworks?page=1&private=true&size=1&sort=-created_at&total_count=true&user_id=...": {
						"time": "0s 481.408ms",
						"cache": false,
						"length": "0 B"
					}
				}
			}
		},
		"requestID": "f9d736a0-b832-11ed-9ec3-970afd3f35b1",
		"userAgent": "insomnia/2022.7.0"
	}
}
```

### CODE REVIEW

File: src/schema/v2/artwork/index.ts [hunk 1]
```diff
@@ -370,6 +377,53 @@ export const ArtworkType = new GraphQLObjectType<any, ResolverContext>({
...
+            offset,
+            total_count: true,
+          })
+          const bodyWithMe = (body ?? []).map((obj) => {
+            return { ...obj, userID: meID }
+          })
+          const totalCount = parseInt(
+            (headers ?? {})["x-total-count"] || "0",
+            10
+          )
```
Comment (dimatretyak) [inline src/schema/v2/artwork/index.ts:415]: **question**: I think `headers` will never be `null` or `undefined`. Or am I wrong?

Comment (dimatretyak) [reply src/schema/v2/artwork/index.ts:415]: **suggestion**: small improvements

```javascript
const totalCount = parseInt(headers?.["x-total-count"] || "0", 10)
```

File: src/schema/v2/artwork/index.ts [hunk 2]
```diff
@@ -370,6 +377,53 @@ export const ArtworkType = new GraphQLObjectType<any, ResolverContext>({
...
+            private: true,
+            default: args.default,
+            saves: args.saves,
+            sort: args.sort,
+            size,
+            offset,
+            total_count: true,
+          })
+          const bodyWithMe = (body ?? []).map((obj) => {
+            return { ...obj, userID: meID }
```
Comment (dimatretyak) [inline src/schema/v2/artwork/index.ts:410]: **question**: why do we need to return `userID` here? Sorry if this is a stupid question 😅

**Review by dimatretyak — APPROVED** [review]
LGTM 👌

File: src/schema/v2/artwork/index.ts (hunk 2 above)
Comment (dimatretyak) [reply src/schema/v2/artwork/index.ts:410]: aaaah, got it
Insomnia 2023-03-01 16-40-44

**Review by anandaroop — APPROVED** [review]

**Pull request merged**